client/hwlib/src/lib.rs

```diff
@@ -19,107 +19,59 @@
-        processor: Some(vec![devices::Processor {
-            family: "Sample Family".to_string(),
-            frequency: 3.5,
-            manufacturer: "Sample Manufacturer".to_string(),
-            version: "v1.0".to_string(),
-        }]),
-        gpu: Some(vec![devices::GPU {
-            family: "Sample Family".to_string(),
-            manufacturer: "Sample Manufacturer".to_string(),
-            version: "v1.0".to_string(),
-        }]),
-        audio: Some(vec![devices::Audio {
-            model: "Sample Model".to_string(),
-            vendor: "Sample Vendor".to_string(),
-        }]),
-        video: Some(vec![devices::VideoCapture {
-            model: "Sample Model".to_string(),
-            vendor: "Sample Vendor".to_string(),
-        }]),
-        network: Some(vec![devices::NetworkAdapter {
-            bus: "Sample Bus".to_string(),
-            id: "Sample ID".to_string(),
-            model: "Sample Model".to_string(),
-            vendor: "Sample Vendor".to_string(),
-            capacity: 1000,
-        }]),
-        wireless: Some(vec![devices::WirelessAdapter {
-            model: "Sample Model".to_string(),
-            vendor: "Sample Vendor".to_string(),
-        }]),
-        pci_peripherals: Some(vec![devices::PCIPeripheral {
-            name: "Sample Name".to_string(),
-            pci_id: "0000:0000".to_string(),
-            vendor: "Sample Vendor".to_string(),
-        }]),
-        usb_peripherals: Some(vec![devices::USBPeripheral {
-            name: "Sample Name".to_string(),
-            usb_id: "0000:0000".to_string(),
-            vendor: "Sample Vendor".to_string(),
-        }]),
-    }
-}
+use constants::CERT_STATUS_ENDPOINT;
+use models::request_validators::CertificationStatusRequest;
+use models::response_validators::{
+    CertificationStatusResponse, CertifiedImageExistsResponse, CertifiedResponse, NotSeenResponse,
+    RelatedCertifiedSystemExistsResponse,
+};
 
-pub async fn get_certification_status(_url: &str) -> Result<serde_json::Value, reqwest::Error> {
-    let response_type = std::env::var("CERTIFICATION_STATUS")
-        .unwrap_or_else(|_| "0".to_string())
-        .parse::<i32>()
-        .unwrap_or(0);
+pub async fn send_certification_request(
+    url: String,
+    request: &CertificationStatusRequest,
+) -> Result<CertificationStatusResponse, Box<dyn std::error::Error>> {
+    let client = Client::new();
+    let mut server_url = url.clone();
+    server_url.push_str(CERT_STATUS_ENDPOINT);
+    let response = client.post(server_url).json(request).send().await?;
 
-    let response = match response_type {
-        1 => serde_json::json!(get_related_certified_system_exists_sample()),
-        2 => serde_json::json!(get_certified_system_sample()),
-        _ => serde_json::json!({"status": "Not Seen"}),
+    let response_text = response.text().await?;
+    let response_value: serde_json::Value = serde_json::from_str(&response_text)?;
+    let parsed_response = match response_value
```

Comment:
There's a way to handle this kind of two-step deserialization using serde that avoids needing to work with the raw `serde_json::Value` structs in this way. It's a little bit fiddly to set up but we've got it working in the apparmor prompting client [like this](https://github.com/canonical/prompting-client/blob/main/prompting-client/src/snapd_client/prompt.rs#L35-L47). (Happy to take you through what's going on there if you want to take a look!)

Reply:
Updated in d7d4680